app/controllers/restaurants_controller.rb

```diff
@@ -19,6 +19,7 @@ def new
 
   def create
     @restaurant = Restaurant.new(restaurant_params)
+    @restaurant.user_id = current_user.id
```

Comment:
Thanks Raoul! Also updated with the comments from Thomas. Hopefully this will work for your comment as well.